lib/Activity.py

```diff
@@ -187,19 +188,38 @@ def load_activity_dict(self, filename=None, encoding='UTF-8'):
 
             aname = list(masterdict)[0]
             activity = masterdict[aname]
+            last_finished = None
 
             self.name = aname
             self.filename = filename
             self.states = dict()
             if "states" in activity:
-                for rtime in activity["states"].keys():
+                for rtime in sorted(activity["states"].keys()):
                     state = activity["states"][rtime]
                     # reprocess the time in case the formats changed or something
                     stime = self.get_time(rtime)
                     self.states[stime] = dict()
                     for attr in ACTIVITY_NEW_STATE:
                         if attr in state:
+                            # Update any sessions that don't have a finished state
+                            if attr == "status" and state[attr] not in ["Succeeded","Failed"]:
```

Comment:
I wonder if when we are processing status states, if it would make sense that we have something that logs if the state is something that we don't know about, like:
```
if ... not in [<list of all known states>]:
  self.config.logger.debug('Recived unknown state {}'.format(state))
```

Reply:
it shouldn't ever happen in this context since it's during reading in the activity yaml, and we do prevent unknown statuses during normal processing.  The only way it'd be wrong here is if someone edits their yaml and I'm reasonably unconcerned about that scenario